inc/namespace.php

```diff
@@ -487,6 +487,14 @@ function get_or_create_wp_user( \OneLogin\Saml2\Auth $saml ) {
 		return new \WP_Error( 'invalid-user', esc_html__( 'Could not create a new user.', 'wp-simple-saml' ) );
 	}
 
+	/**
+	 * Used to handle post-user-match logic, ie: role mapping, updating attributes.
+	 *
+	 * @param \WP_User $user       User object
+	 * @param array    $attributes SAML Attributes passed from IdP
+	 */
+	do_action( 'wpsimplesaml_user_matched', $user, $attributes );
```

Comment:
What's the functional difference in this PR? It's not clear to me why the existing action is not good enough here.

Reply:
This one acts on each authentication, the other action only works when a user account is initially created.

Changes to settings in active directory once a user has logged in once will never be reflected in WordPress unless the roles mapping is checked again on each authentication.

Reply:
This seems like a fairly major change in behaviour that might have subtle effects; is there an issue where we can discuss that behaviour in more depth?